esrally/client/factory.py

```diff
@@ -31,7 +31,7 @@ class EsClientFactory:
     compatibility guarantees that are broader than the library's defaults.
     """
 
-    def __init__(self, hosts, client_options, distribution_version=None):
+    def __init__(self, hosts, client_options, distribution_version=None, distribution_flavor=None):
```

Comment:
Why do we accept `None` values here?

Reply:
The factory is also called before we know what the distribution version/flavor is, e.g:
https://github.com/elastic/rally/blob/abf43207e1ebde6899876dcc3ebadf5998e7eab1/esrally/client/factory.py#L326-L349